RELEASE/scripts/autoscend/quests/level_any.ash

```diff
@@ -697,3 +697,142 @@ boolean LX_meatMaid()
 
 	return false;
 }
+
+string LX_getDesiredWorkshed(){
+	string currentWorkshed = get_property("auto_workshed").to_lower_case();
+	//return the actual item name in case a shorthand is used
+	switch(currentWorkshed)
+	{
+		case "train":
+			return "model train set";
+		case "model train set":
+		case "cmc":
+			return "cold medicine cabinet";
+		case "cold medicine cabinet":
+		case "pizza": //unsupported
+			return "diabolic pizza cube";
+		case "diabolic pizza cube": //unsupported
+		case "asdon":
+			return "Asdon Martin keyfob";
+		case "Asdon Martin keyfob":
+		case "mayo":
+			return "portable mayo clinic";
+		case "portable mayo clinic":
+		case "dnalab":
+			return "little geneticist dna-splicing lab";
+		case "little geneticist dna-splicing lab":
+		case "snow machine": //unsupported
+		case "spinning wheel": //unsupported
+		case "warbear auto-anvil": //unsupported
+		case "warbear chemistry lab": //unsupported
+		case "warbear high-efficiency still": //unsupported
+		case "warbear induction oven": //unsupported
+		case "warbear jackhammer drill press": //unsupported
+		case "warbear lp-rom burner": //unsupported
+		default:
+			// auto_workshed is invalid or none/false/whatever to say don't do this
+			return "auto";
+	}
+}
+
+boolean LX_setWorkshed(){
+	string desiredShed = LX_getDesiredWorkshed();
+	string existingShed = get_workshed();
+
+	//set up an integer to make workshed check simpler especially if new worksheds are added
+	//later we check once train set's usefulness is done whether we can use Asdon or not
+	//if we can't we move to the next item. If we can, as long as existingShedInt does not equal 2,
+	//we would want to have the CMC installed. If, the user doesn't have, either of the top 3, we use
+	//the lab or mayo if the user doesn't have lab.
+	int existingShedInt = 0;
+	if (existingShed == "model train set")
+	{
+		existingShedInt = 1;
+	}
+	else if (existingShed == "Asdon Martin keyfob")
+	{
+		existingShedInt = 2;
+	}
+	else if (existingShed == "cold medicine cabinet")
+	{
+		existingShedInt = 3;
+	}
+	else if (existingShed == "little geneticist dna-splicing lab")
+	{
+		existingShedInt = 4;
+	}
+	else if (existingShed == "portable mayo clinic")
+	{
+		existingShedInt = 5;
+	}
+	//Check to make sure we can use the workshed item and that it isn't already in the campground. If already in campground, return true also
+	if (auto_is_valid(to_item(desiredShed)) && (contains_text(existingShed, desiredShed) == false) && (item_amount(to_item(desiredShed)) > 0))
+	{
+		use(1, to_item(desiredShed));
+		return true;
+	}
+	if (contains_text(existingShed, desiredShed))
+	{
+		return false;
+	}
+	if ((contains_text(desiredShed,"auto")) && (fastenerCount() < 30 && lumberCount() < 30) && (existingShed == "none"))

```

Comment:
Started looking at this if structure, but unfortunately have ran out of time. So just going to give ya a short note

First inclination is to try to simplify this. The order of if statements will impose a priority of which workshop is better than others. So I think the conversion to ints can be removed. Adds code and makes reading this if statement a little harder

What we talked about in ASS was to first go to train, if you have it, and then go to the next highest priority. I'd add an if statement above to return false if workshed != none and != train. That way we don't need to worry about preventing more than one workshed change. If you do this, I believe you can just have a single if else chain in the priority of worksheds... haven't given it enough time to truly think it through though

Will come back to this early next week. It will be a busy weekend